llama_index/readers/metal.py

```diff
@@ -0,0 +1,75 @@
+from typing import Any, Dict, List, Optional
+
+from gpt_index.readers.base import BaseReader
+from gpt_index.readers.schema.base import Document
+
+
+class MetalReader(BaseReader):
+    """Metal reader.
+
+    Args:
+        api_key (str): Metal API key.
+        client_id (str): Metal client ID.
+        index_id (str): Metal index ID.
+    """
+
+    def __init__(self, api_key: str, client_id: str, index_id: str):
+        import_err_msg = (
+            "`metal_sdk` package not found, please run `pip install metal_sdk`"
+        )
+        try:
+            import metal_sdk  # noqa: F401
+        except ImportError:
+            raise ImportError(import_err_msg)
+        from metal_sdk.metal import Metal
+
+        """Initialize with parameters."""
+        self._api_key = api_key
+        self._client_id = client_id
+        self._index_id = index_id
+        self.metal_client = Metal(api_key, client_id, index_id)
+
+    def load_data(
+        self,
+        limit: int,
+        id_to_text_map: Optional[Dict[str, str]],
+        query_embedding: Optional[List[float]] = None,
+        filters: Optional[Dict[str, Any]] = None,
+        separate_documents: bool = True,
+        **query_kwargs: Any
+    ) -> List[Document]:
+        """Load data from Metal.
+
+        Args:
+            id_to_text_map (Dict[str, str]): A map from ID's to text.
+            query_embedding (Optional[List[float]]): Query embedding for search.
+            limit (int): Number of results to return.
+            filters (Optional[Dict[str, Any]]): Filters to apply to the search.
+            separate_documents (Optional[bool]): Whether to return separate
+                documents per retrieved entry. Defaults to True.
+            **query_kwargs: Keyword arguments to pass to the search.
+
+        Returns:
+            List[Document]: A list of documents.
+        """
+
+        payload = {
+            "embedding": query_embedding,
+            "filters": filters,
+        }
+        response = self.metal_client.search(payload, limit=limit, **query_kwargs)
+
+        documents = []
+        for item in response["data"]:
+            text = item["text"]
+            if id_to_text_map is not None and item["id"] in id_to_text_map:
```

Comment:
@Disiok i still don't fully understand the need for this ..is this standard in llamaindex to add expected results to `id_to_text_map` before hitting this method? we pass back `text` from our api

Reply:
Hey @softboyjimbo, I think `id_to_text_map` was introduced to handle vector DBs that don't store text (my best guess).
It's not necessary here. So let's just remove the param entirely.